meshack-mbuvi/ride-my-way-api
# Bg fix action on user request 158831144

#### What does this PR do?
Fixes bug where a driver is able to accept user request more than once for the same user

#### Description of Task to be completed?
- Query database to check for the status of the user request.
- Process driver's action to accept or reject ride offer depending on the status obtained from the database

#### How should this be manually tested?
using your terminal,
- clone this repo
- `cd ride-my-way`
- set and activate your virtual environment
- `git checkout bg-fix-action-on-user-request-158831144`
- Run `nosetests -v` to confirm all tests run successfully
- Run `python3 run.py`
- Using postman, 
    ```
    - create a ride offer as a driver
    - request to join the ride as a passenger
    - log in as a driver and accept the user request more than once.

**Pivotal tracker stories**
[Fix driver's action to accept/ reject user request](https://www.pivotaltracker.com/story/show/158804740)
